parsestruct.go

```diff
@@ -0,0 +1,207 @@
+package rig
+
+import (
+	"flag"
+	"net/url"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type fieldInfo struct {
+	field reflect.Value
+	typ   reflect.StructField
+
+	flag     string
+	env      string
+	usage    string
+	typeHint string
+	required bool
+
+	isStruct bool
+}
+
+func parseBool(s string) (bool, error) {
+	if s == "" {
+		return false, nil
+	}
+
+	return strconv.ParseBool(s)
+}
+
+func getFieldInfo(field reflect.Value, typ reflect.StructField) (*fieldInfo, error) {
+	ignore, err := parseBool(typ.Tag.Get("ignore"))
+	if err != nil || ignore {
```

Comment:
`ignore` is still under debate, but we shouldn't return an error if something is ignored. I'm not sure if `err` can be anything but `nil` when `ignore = true`, but if it can, this becomes unintuitive.